src/components/Checkbox/Checkbox.scss

```diff
@@ -1,82 +1,121 @@
 @import '../../styles/common';
+$control-size: rem(16px);
 
 .Checkbox {
   position: relative;
-}
-
-.error {
-  // The basic checked styles have more specificity than this, so need to
-  // basically duplicate that selector here, too.
-  // stylelint-disable-next-line selector-max-specificity
-  .Input:checked + .Backdrop,
-  .Input.Input-indeterminate + .Backdrop,
-  .Backdrop {
-    @include control-backdrop(error);
-  }
-
-  // We need to know if the input is focusing to change the border color.
-  // stylelint-disable-next-line selector-max-specificity
-  .Input:focus + .Backdrop,
-  .Input:active + .Backdrop {
-    @include control-backdrop(focused-error);
-  }
-
-  .Icon {
-    @include recolor-icon(color('red', 'dark'));
-  }
+  margin: var(--p-choice-margin, 0);
 }
 
 .Input {
   @include visually-hidden;
 
-  &:focus,
-  &:active,
-  &:checked,
-  &.Input-indeterminate {
-    // stylelint-disable-next-line selector-max-class
-    + .Backdrop {
-      background: color('indigo');
+  &:not(.globalTheming) {
```

Comment:
Kept the `.globalTheming` and `:not(.globalTheming)` completely separate. Because these are quite different this will make it much easier to remove the old code once this is complete.